ibis/common/scope.py

```diff
@@ -0,0 +1,149 @@
+""" Module for scope
+    The motivation of Scope is to cache data for calculated ops.
+
+    `scope` in Scope class is the main cache. It is a dictionary mapping
+    ibis node instances to concrete data, and the time context associate
+    with it(if any).
+
+    `scope` has the structure of:
```

Comment:
I think doc specific about implementation the Scope class can be moved into the class it self.

Module doc should be high level.

Reply:
Just removed it for now.